### PR TITLE
Email instances for Infra VM provisioning.

### PR DESCRIPTION
**Pull request by billfitzgerald0120**
Added 7 instances in System/Notification/Email class for Infra/VM provisioning.

Modified EmailOwner value in 2 State Machine classes and clone_to_vm instance to
use new instances.

/Infrastructure/VM/Provisioning/StateMachines/VMProvision_Template.class/class.yaml
/Infrastructure/VM/Provisioning/StateMachines/VMProvision_VM.class/class.yaml
/Infrastructure/VM/Provisioning/StateMachines/VMProvision_VM.class/clone_to_vm.yaml

https://bugzilla.redhat.com/show_bug.cgi?id=1314871
https://www.pivotaltracker.com/epic/show/3861726

### CODE REVIEW

File: content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/VMProvision_VM.class/clone_to_vm.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ object:
   - PreProvision:
       value: "/Infrastructure/VM/Provisioning/StateMachines/Methods/PreProvision_Clone_to_VM#${/#miq_provision.source.vendor}"
   - EmailOwner:
-      value: "/Infrastructure/VM/Provisioning/Email/MiqProvision_Complete?event=vm_cloned"
+      value: "/System/Notification/Email/InfrastructureMiqProvisionComplete?event=vm_cloned"
```
Comment (mkanoor) [inline content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/VMProvision_VM.class/clone_to_vm.yaml]: @tinaafitz 
do you know if the event is needed when calling the method, is it used by the called instance or any other child of the called instance.

File: content/automate/ManageIQ/System/Notification/Email.class/infrastructuremiqprovisionrequestapproverapproved.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (mkanoor) [inline content/automate/ManageIQ/System/Notification/Email.class/infrastructuremiqprovisionrequestapproverapproved.yaml]: @billfitzgerald0120 
Is there a state machine step that calls this instance, should that be attached to this PR?

File: content/automate/ManageIQ/System/Notification/Email.class/infrastructuremiqprovisioncomplete.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    description: 
+  fields:
+  - to:
+      value: "${/#miq_provision.miq_request.get_option(:owner_email)} || ${/#miq_provision.miq_request.requester.email}
+        || ${/Configuration/Email/Default#default_recipient}"
+  - subject:
+      value: 'Request ID ${/#miq_provision.miq_request.id} - Your Virtual Machine
+        Request has Completed - Vm Name: <${/#miq_provision.vm.name}>'
+  - customize:
+      value: miq_provision_complete
```
Comment (mkanoor) [inline content/automate/ManageIQ/System/Notification/Email.class/infrastructuremiqprovisioncomplete.yaml]: @billfitzgerald0120 
Should this be the new customize body method?

**Comment by billfitzgerald0120** [comment]
@miq-bot add_label enhancement

**Pull request merged**